ArcGISRuntimeSDKQt_QMLSamples/Maps/GenerateOfflineMapLocalBasemap/GenerateOfflineMapLocalBasemap.qml

```diff
@@ -0,0 +1,269 @@
+// [WriteFile Name=GenerateOfflineMapLocalBasemap, Category=Maps]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.6
+import QtQuick.Controls 2.2
+import QtQuick.Layouts 1.3
+import Esri.ArcGISRuntime 100.6
+import Esri.ArcGISExtras 1.1
+import Esri.ArcGISRuntime.Toolkit.Dialogs 100.6
+
+Rectangle {
+    id: rootRectangle
+    clip: true
+    width: 800
+    height: 600
+
+    property url outputMapPackage: System.temporaryFolder.url + "/OfflineMap_%1.mmpk".arg(new Date().getTime().toString())
+    property url basemapDirectory: System.userHomePath + "/ArcGIS/Runtime/Data/tpk"
+    property string webMapId: "acc027394bc84c2fb04d1ed317aac674"
+    property bool useLocalBasemap: false
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+        // Create a Map from a Portal Item
+        Map {
+            id: map
+
+            PortalItem {
+                id: mapPortalItem
+                itemId: webMapId
+            }
+        }
+
+        // Create a button and anchor it to the attribution top
+        DownloadButton {
+            id: downloadButton
+            anchors {
+                horizontalCenter: parent.horizontalCenter
+                bottom: mapView.attributionTop
+                margins: 5
+            }
+            visible: map.loadStatus === Enums.LoadStatusLoaded
+
+            onButtonClicked: dialog.open()
+        }
+    }
+
+    // Create an extent rectangle for selecting the offline area
+    Rectangle {
+        id: extentRectangle
+        anchors.centerIn: parent
+        width: parent.width - (50)
+        height: parent.height - (125)
```

Comment:
I don't see why these numbers need to be parenthesized.

Reply:
updated. this is a remnant of our scaleFactor removal find & replace. there are probably a few more instances of these. i cleaned up the 3 generate offline map samples